include/deal.II/base/symmetric_tensor.h

```diff
@@ -1068,6 +1084,289 @@ namespace internal
+        tmp[idx_11] = t[idx_00];
+        tmp *= inv_det_t;
+
+        return tmp;
+      }
+    };
+
+
+    template<typename Number>
+    struct Inverse<2,3,Number>
+    {
+      static dealii::SymmetricTensor<2,3,Number>
+      value (const dealii::SymmetricTensor<2,3,Number> &t)
+      {
+        dealii::SymmetricTensor<2,3,Number> tmp;
+
+        // Sympy result: ([
+        // [  (t11*t22 - t12**2)/(t00*t11*t22 - t00*t12**2 - t01**2*t22 + 2*t01*t02*t12 - t02**2*t11),
+        //    (-t01*t22 + t02*t12)/(t00*t11*t22 - t00*t12**2 - t01**2*t22 + 2*t01*t02*t12 - t02**2*t11),
+        //    (t01*t12 - t02*t11)/(t00*t11*t22 - t00*t12**2 - t01**2*t22 + 2*t01*t02*t12 - t02**2*t11)],
+        // [  (-t01*t22 + t02*t12)/(t00*t11*t22 - t00*t12**2 - t01**2*t22 + 2*t01*t02*t12 - t02**2*t11),
+        //    (t00*t22 - t02**2)/(t00*t11*t22 - t00*t12**2 - t01**2*t22 + 2*t01*t02*t12 - t02**2*t11),
+        //    (t00*t12 - t01*t02)/(-t00*t11*t22 + t00*t12**2 + t01**2*t22 - 2*t01*t02*t12 + t02**2*t11)],
+        // [  (t01*t12 - t02*t11)/(t00*t11*t22 - t00*t12**2 - t01**2*t22 + 2*t01*t02*t12 - t02**2*t11),
+        //    (t00*t12 - t01*t02)/(-t00*t11*t22 + t00*t12**2 + t01**2*t22 - 2*t01*t02*t12 + t02**2*t11),
+        //    (-t00*t11 + t01**2)/(-t00*t11*t22 + t00*t12**2 + t01**2*t22 - 2*t01*t02*t12 + t02**2*t11)]  ])
+        const TableIndices<2> idx_00 (0,0);
+        const TableIndices<2> idx_01 (0,1);
+        const TableIndices<2> idx_02 (0,2);
+        const TableIndices<2> idx_11 (1,1);
+        const TableIndices<2> idx_12 (1,2);
+        const TableIndices<2> idx_22 (2,2);
+        const Number inv_det_t
+          = 1.0/(t[idx_00]*t[idx_11]*t[idx_22]
+                 - t[idx_00]*t[idx_12]*t[idx_12]
+                 - t[idx_01]*t[idx_01]*t[idx_22]
+                 + 2.0*t[idx_01]*t[idx_02]*t[idx_12]
+                 - t[idx_02]*t[idx_02]*t[idx_11]);
+        tmp[idx_00] = t[idx_11]*t[idx_22] - t[idx_12]*t[idx_12];
+        tmp[idx_01] = -t[idx_01]*t[idx_22] + t[idx_02]*t[idx_12];
+        tmp[idx_02] = t[idx_01]*t[idx_12] - t[idx_02]*t[idx_11];
+        tmp[idx_11] = t[idx_00]*t[idx_22] - t[idx_02]*t[idx_02];
+        tmp[idx_12] = -t[idx_00]*t[idx_12] + t[idx_01]*t[idx_02];
+        tmp[idx_22] = t[idx_00]*t[idx_11] - t[idx_01]*t[idx_01];
+        tmp *= inv_det_t;
+
+        return tmp;
+      }
+    };
+
+
+    template<typename Number>
+    struct Inverse<4,1,Number>
+    {
+      static inline dealii::SymmetricTensor<4,1,Number>
+      value (const dealii::SymmetricTensor<4,1,Number> &t)
+      {
+        dealii::SymmetricTensor<4,1,Number> tmp;
+        tmp.data[0][0] = 1.0/t.data[0][0];
+        return tmp;
+      }
+    };
+
+
+    template<typename Number>
+    struct Inverse<4,2,Number>
+    {
+      static inline dealii::SymmetricTensor<4,2,Number>
+      value (const dealii::SymmetricTensor<4,2,Number> &t)
+      {
+        dealii::SymmetricTensor<4,2,Number> tmp;
+
+        // inverting this tensor is a little more
```

Comment:
i see you copy-paste it from below,but you can also adjust the comments to be a bit larger as we do elsewhere